var/spack/repos/builtin/packages/plumed/package.py

```diff
@@ -87,6 +116,29 @@ class Plumed(AutotoolsPackage):
     depends_on('m4', type='build')
     depends_on('py-cython', type='build', when='@2.5:')
 
+    # Specific optional modules only added on top of `optional_modules` = none or reset
+    for mod in single_optional_modules:
+        conflicts('+' + mod, when='optional_modules=all', msg='specific optional modules require optional_modules=none or reset')
+
+    # Conflicts between PLUMED versions and specific optional modules
+    conflicts('+imd', when='@2.3:', msg='imd was removed from version 2.3')
+    conflicts('+reference', when='@2.3:', msg='reference was removed from version 2.3')
+    conflicts('+vesselbase', when='@2.3:', msg='vesselbase was removed from version 2.3')
+    conflicts('+adjmat', when='@:2.2.99', msg='adjmat was added from version 2.3')
+    conflicts('+drr', when='@:2.3.99', msg='drr was added from version 2.4')
+    conflicts('+eds', when='@:2.3.99', msg='eds was added from version 2.4')
+    conflicts('+isdb', when='@:2.3.99', msg='isdb was added from version 2.4')
+    conflicts('+ves', when='@:2.3.99', msg='ves was added from version 2.4')
+    conflicts('+dimred', when='@:2.4.99', msg='dimred was added from version 2.5')
+    conflicts('+logmfd', when='@:2.4.99', msg='logmfd was added from version 2.5')
+    conflicts('+pamm', when='@:2.4.99', msg='pamm was added from version 2.5')
+    conflicts('+piv', when='@:2.4.99', msg='piv was added from version 2.5')
+    conflicts('+annfunc', when='@:2.5.99', msg='annfunc was added from version 2.6')
+    conflicts('+maze', when='@:2.5.99', msg='maze was added from version 2.6')
+    conflicts('+fisst', when='@:2.6.99', msg='fisst was added from version 2.7')
+    conflicts('+funnel', when='@:2.6.99', msg='funnel was added from version 2.7')
+    conflicts('+opes', when='@:2.6.99', msg='opes was added from version 2.7')
```

Comment:
This can be rewritten using #29530 and a multi-valued variant. You can check `nsimd` for an example of that.

Reply:
This now seems resolved to me @alalazo